process_report/process_report.py

```diff
@@ -1,9 +1,108 @@
 import argparse
 import os
+import sys
 
 import pandas
 
 
+### Invoice field names
+INVOICE_MONTH_FIELD = 'Invoice Month'
+PROJECT_FIELD = 'Project - Allocation'
+MANAGER_PI_FIELD = 'Manager (PI)'
+INSTITUTION_FIELD = 'Institution'
+SU_HOUR_FIELD = 'SU Hours (GBhr or SUhr)'
+SU_TYPE_FIELD = 'SU Type'
+COST_FIELD = 'Cost'
+CREDIT_FIELD = 'Credit'
+CREDIT_CODE_FIELD = 'Credit Code'
+BALANCE_FIELD = 'Balance'
+###
+
+
+def apply_credits_0001(dataframe):
+    credit_code = "0001"
+    pass
+
+
+def apply_credits_0002(dataframe):
+    """Applies the New PI Credit. This credit function expects the 
+    env var `C0002_OLD_PI` to be set, pointing to a txt file containing old PIs"""
+    credit_code = "0002"
+    credit_amount = 1000
+
+    old_pi_list = set()
+    try:
+        with open(os.getenv("C0002_OLD_PI")) as f:
+            for pi in f: old_pi_list.add(pi.strip())
+
+    except Exception:
+        print("Old PI file does not exist, or the C0002_OLD_PI env var is not set")
+        sys.exit(1)
+
+    pi_list = dataframe[MANAGER_PI_FIELD].unique()
+
+    for pi in pi_list:
+        if pi != pi: continue   # NaN check
```

Comment:
can you say more about this check? The comment is not helpful.

Reply:
Also, it's not a popular to write if/esle statements in one line. Please stick to the standard formatting. You could use [`black`](https://github.com/psf/black) to format your code; although it formats the whole file, so don't use it in this PR as it'll make the diff unreadable. Kust manually fix the if/else statements please.

Reply:
@naved001 While running tests on past invoice CSVs, I found that some project allocations did not have a value in their PI field. When `pandas` read the invoice CSV, it assigns those empty fields with the value `NaN`. Using `if pi != pi` was one way I found to check whether the PI field was empty or not. I've updated my PR in #12 to use `pandas.isna()`, which should make the intention more clear.